### ISSUE TITLE
Matplotlib is not Unyt aware.

### ISSUE THREAD

**Issue by l-johnston** [yt-project/unyt#121]
* unyt version: 2.4.1
* Python version: 3.8.1
* Operating System: Windows 10

### Feature proposal:
I propose making Matplotlib plots Unyt aware. As is, a plot generated from unyt_array does not make use of the unyt_array features such as the unit label or unit conversions. For example:
```python
>>> import matplotlib.pyplot as plt
>>> from unyt import s, K
>>> x = [0.0, 0.01, 0.02]*s
>>> y = [298.15, 308.15, 318.15]*K
>>> plt.plot(x, y)
[<matplotlib.lines.Line2D object at ...>]
>>> plt.show()
```
![fig0](https://user-images.githubusercontent.com/17813788/72636910-c0d55100-3925-11ea-9eee-1a56128e1ae6.png)

Matplotlib offers a ConversionInterface to process custom types such as Unyt. I have implemented this interface in [my fork](https://github.com/l-johnston/unyt/tree/mpl_interface). With this interface, the above code produces the plot:
![fig1](https://user-images.githubusercontent.com/17813788/72637271-8c15c980-3926-11ea-8090-a41bc9a7325d.png)

It is also possible to convert the units:
```python
>>> plt.plot(x, y, xunits="ms", yunits=("zJ", "thermal"))
>>> plt.show()
```
![fig2](https://user-images.githubusercontent.com/17813788/72637402-d7c87300-3926-11ea-9669-085e9325a447.png)

Is the Unyt team interested in this feature?

**Comment by ngoldbaum** [comment]
Sure!

You'll need to add matplotlib to the test dependencies and make sure the tox tests install matplotlib and run the new tests. New code also needs to have 100 percent test coverage.

You'll also need to add documentation for this feature.

Let me know if you have questions about testing, documentation, or development workflow.

**Comment by l-johnston** [comment]
In tox.ini, do I add matplotlib under each [testenv...]?

Should I place the documentation in the usage.rst file after the "Integrating unyt into a Python Library" section? And titled "Plotting with Matplotlib"?

**Comment by ngoldbaum** [comment]
> In tox.ini, do I add matplotlib under each [testenv...]?

Yup, except for the `py35-dependencies` job, which tests a minimal set of dependencies, and the matplotlib support should be optional so matplotlib shouldn't get installed in that environment.

For the `py35-versions` job, pick the minimum version of matplotlib you think we should support. This probably should be whichever matplotlib version this functionality works in or some reasonably later version, not the most recent version.

If you want to run the tox tests locally you'll need to have python 3.5, 3.6, 3.7, and 3.8 executables simultaneously installed. I do this with `pyenv`. If you don't have `pyenv` or don't want to deal with installing all the different python versions you can just run tox with one python version and the testenvs for the other versions will fail. You rely on travis to run the tests as well if you don't mind the tests not running locally.

> Should I place the documentation in the usage.rst file after the "Integrating unyt into a Python Library" section? And titled "Plotting with Matplotlib"?

Sure that's probably as good a place as any. Use your best judgement. Also try to build the docs locally after you modify them to make sure they look OK.

**Comment by ngoldbaum** [comment]
closed by #122

**Comment by JBorrow** [comment]
I remember pushing back on this feature a little at the time, but I'd just like to say that now it really helps, and with all of the extra changes (the context manager, ability to use square brackets), it's one of my favourite things about unyt. All of our internal libraries now respect this, along with the `name` field, to produce plots that would take a whole script in just a few lines.
